src/generic/Matrix.jl

```diff
@@ -2925,7 +2950,7 @@ function minpoly(S::Ring, M::MatElem{T}, charpoly_only::Bool = false) where {T <
    first_poly = true
    while r2 <= n
       P1 = [0 for i in 1:2n + 1]
-      v = similar(M, n, 1)
+      v = zero_matrix(R, n, 1)
```

Comment:
Those are not equivalent.

It would be good to have a `zero(::MatrixElem, ::Int, ::Int)` function, exactly for this purpose here.